### PR TITLE
removed blackduck version from CRD defaults

### LINKED ISSUE #227: crd defaults should not include a blackduck version
The operator automatically installs the latest version if no version is specified.

### PR DESCRIPTION
**Pull request by VSharapov**
Fixes #227 

[Tested](https://github.com/VSharapov/synopsys-operator/tree/buildable) on top of [the last buildable commit](https://github.com/blackducksoftware/synopsys-operator/commit/cb98a61fe33a35c1e988fe725fa8c4c9e895243a) and seems to spin up a ~~hub~~ blackduck without issue.

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/369#" title="Author self-approved">VSharapov</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **mphammer**

If they are not already assigned, you can assign the PR to them by writing `/assign @mphammer` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/util/crd_defaults.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func GetBlackDuckTemplate() *blackduckv1.BlackduckSpec {
 		DbPrototype:       "",
 		CertificateName:   "default",
 		Type:              "",
-		Version:           "2019.4.2",
+		Version:           "",
```
Comment (msenmurugan) [inline pkg/util/crd_defaults.go]: Probably we can remove all the empty fields from the struct. Default value for String is "".

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/369#issuecomment-495593528" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/369#" title="Author self-approved">VSharapov</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/369#pullrequestreview-241700000" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/369#" title="Author self-approved">VSharapov</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@VSharapov: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by VSharapov** [comment]
@msenmurugan 

> Probably we can remove all the empty fields from the struct. Default value for String is "".

All structs in the crd_defaults?

**Pull request closed without merging**